### PR TITLE
Feature: telemetry module

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

1. Encapsulate `record.EventRecorder` interface in k8s client-go package, upload telemetry data alongside operator manager create event. 
2. A throttler which hold a buffered channel and a group of workers who fetch uploading data in infinite loop.
3. A metrics puller which extract informations from the host environment
4. A flexible asynchronous method to generate telemetry record and send to the buffered channel of throttler
5. Disable or debug the module with environment variables
6. All functions pass bundled unit tests

### CODE REVIEW

File: pkg/controller/obtenantbackup_controller.go [hunk 1]
```diff
@@ -41,8 +44,11 @@ import (
 // OBTenantBackupReconciler reconciles a OBTenantBackup object
 type OBTenantBackupReconciler struct {
 	client.Client
-	Scheme   *runtime.Scheme
-	Recorder record.EventRecorder
+	Scheme    *runtime.Scheme
+	Recorder  record.EventRecorder
+	Telemetry telemetry.Recorder
```
Comment (chris-sun-star) [inline pkg/controller/obtenantbackup_controller.go:49]: use telemetry as Recorder

File: pkg/resource/obcluster_task.go [hunk 2]
```diff
@@ -820,5 +827,6 @@ func (m *OBClusterManager) RestoreEssentialParameters() error {
 		}
 	}
 	_ = m.Client.Delete(m.Ctx, contextSecret)
+	m.Recorder.Event(m.OBCluster, "Upgrade", "", "Restore essential parameters successfully")
```
Comment (chris-sun-star) [inline pkg/resource/obcluster_task.go:830]: this is not necessary

File: pkg/controller/obtenantbackup_controller.go [hunk 3]
```diff
@@ -44,9 +43,8 @@ import (
 // OBTenantBackupReconciler reconciles a OBTenantBackup object
 type OBTenantBackupReconciler struct {
 	client.Client
-	Scheme    *runtime.Scheme
-	Recorder  record.EventRecorder
-	Telemetry telemetry.Recorder
+	Scheme   *runtime.Scheme
+	Recorder telemetry.Recorder
 
 	telemetryOnce sync.Once
```
Comment (chris-sun-star) [inline pkg/controller/obtenantbackup_controller.go:49]: delete telemetryOnce

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Comment by PowerfooI** [comment]
Add event recordings for important transitions. #38

**Pull request merged**